### PR TITLE
improve readme for contributor experience

### PR DESCRIPTION
**Pull request by sebgoa**
**What this PR does / why we need it**:

This improves the top-level README. Removes broken links, removes links to make you bounce to another page, adds proper links to docs.
Shuffles things around a bit and improves reading.

@kubernetes/contributor-experience

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36817)

### CODE REVIEW

**Comment by metral** [comment]
This PR removes links to external community efforts around k8s, e.g. Awesome-Kubernetes & CoreKube etc. - (I started CoreKube).

Is it worth it to place these efforts in a separate page dedicated for community resources as folks still find these sorts of efforts helpful?

**Comment by sebgoa** [comment]
To plead my case, as I mentioned on twitter, I was doing a review of the README and when I hit the bottom I saw those links. CoreKube is really nice but the collections page is empty.

I pondered a lot on what to do with them, and then decided to remove them.

I think they would be better on the website. there is a big community page that needs lots of love. There is also a big Tools page that needs love as well.

**Comment by metral** [comment]
Makes total sense 👍 .

I'll defer to Sarah to suggest a more appropriate home for these sorts of pages.

**Comment by bgrant0607** [comment]
cc @kubernetes/docs

**Comment by bgrant0607** [comment]
Copying from slack:

I don't think we should link to community resources from kubernetes/README.md. We should find a place on kubernetes.io for them. 

Files to remove:
* CONTRIB.md
* DESIGN.md
* contrib/README.md
* docs/warning.png
* docs/roadmap.md
* docs/troubleshooting.md
* docs/whatisk8s.md
* www/README.md

File: README.md [hunk 1]
```diff
@@ -2,22 +2,19 @@
...
+## Introduction
 
-<hr>
-
-Kubernetes is an open source system for managing [containerized applications](https://github.com/kubernetes/kubernetes/wiki/Why-Kubernetes%3F#why-containers) across multiple hosts,
-providing basic mechanisms for deployment, maintenance, and scaling of applications.
+Kubernetes is an open source system for managing [containerized applications](http://kubernetes.io/docs/whatisk8s/) across multiple hosts,
+providing basic mechanisms for deployment, maintenance, and scaling of applications. Kubernetes is hosted by the Cloud Native Computing Foundation ([CNCF](https://www.cncf.io))
 
 Kubernetes is:
```
Comment (bgrant0607) [inline README.md]: Please remove these bullets.

File: README.md [hunk 2]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
 * **extensible**: modular, pluggable, hookable, composable
 * **self-healing**: auto-placement, auto-restart, auto-replication
 
-Kubernetes builds upon a [decade and a half of experience at Google running production workloads at scale](https://research.google.com/pubs/pub43438.html), combined with best-of-breed ideas and practices from the community.
+Kubernetes builds upon a decade and a half of experience at Google running production workloads at scale using a system called [Borg](https://research.google.com/pubs/pub43438.html), combined with best-of-breed ideas and practices from the community.
+
+<hr>
+
+### Kubernetes is ready for Production !
```
Comment (bgrant0607) [inline README.md]: I'd just remove this subsection.

File: README.md [hunk 3]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
+
+Since the Kubernetes 1.0 release in July 2015 Kubernetes is ready for your production workloads. Check the [case studies](http://kubernetes.io/case-studies/).
+
+### Kubernetes can run anywhere !
```
Comment (bgrant0607) [inline README.md]: I'd just remove this subsection.

File: README.md [hunk 4]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
+
+You can run Kubernetes on your local workstation, cloud providers (e.g. GCE, AWS, Azure), on-premises virtual machines and physical hardware. Essentially, anywhere Linux runs you can run Kubernetes. Checkout the [deployment solutions](http://kubernetes.io/docs/getting-started-guides/) for details.
 
 <hr>
 
-### Kubernetes is ready for Production!
+### Are you ...
+
+  * Interested in learning more about using Kubernetes?  Please see our user-facing documentation on [kubernetes.io](http://kubernetes.io). Try our [interactive tutorial](http://kubernetes.io/docs/tutorials/kubernetes-basics/) or take a free course on [Scalable Microservices with Kubernetes](https://www.udacity.com/course/scalable-microservices-with-kubernetes--ud615).
+  * Interested in hacking on the core Kubernetes code base, developing tools using the Kubernetes API or helping in anyway possible ?  Keep reading!
```
Comment (bgrant0607) [inline README.md]: or helping with the project

File: README.md [hunk 5]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
+## Code of Conduct
 
-### Kubernetes can run anywhere!
+The Kubernetes community abides by the CNCF [code of conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md). Here is an excerpt:
 
-You can run Kubernetes on your local workstation under Vagrant, cloud providers (e.g. GCE, AWS, Azure), and physical hardware. Essentially, anywhere Linux runs you can run Kubernetes. Checkout the [Getting Started Guides](http://kubernetes.io/docs/getting-started-guides/) for details.
+_As contributors and maintainers of this project, and in the interest of fostering an open and welcoming community, we pledge to respect all people who contribute through reporting issues, posting feature requests, updating documentation, submitting pull requests or patches, and other activities._
 
-## Concepts
+## Concepts Overview
```
Comment (bgrant0607) [inline README.md]: Please remove this subsection.

File: README.md [hunk 6]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
 
-[**Label**](docs/user-guide/labels.md)
+[**Label**](http://kubernetes.io/docs/user-guide/labels/)
 : Labels are used to organize and select groups of objects based on key:value pairs.
 
-## Documentation
+## Community
 
-Kubernetes documentation is organized into several categories.
+Do you want to help " shape the evolution of technologies that are container packaged, dynamically scheduled and microservices oriented? ". If you are a company, you should consider joining the [CNCF](https://cncf.io/about). For details about who's involved and how Kubernetes plays a role, read [the announcement](https://cncf.io/news/announcement/2015/07/new-cloud-native-computing-foundation-drive-alignment-among-container).
```
Comment (bgrant0607) [inline README.md]: container-packaged
microservices-oriented

English is right-associative.

File: README.md [hunk 7]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
-You should consider joining the [Cloud Native Computing Foundation](https://cncf.io/about). For details about who's involved and how Kubernetes plays a role, read [their announcement](https://cncf.io/news/announcement/2015/07/new-cloud-native-computing-foundation-drive-alignment-among-container).
-
-### Code of conduct
-
-Participation in the Kubernetes community is governed by the [Kubernetes Code of Conduct](code-of-conduct.md).
-
-### Are you ready to add to the discussion?
-
-We have presence on:
+Join us on social media and read our blog:
```
Comment (bgrant0607) [inline README.md]: I'd prefer to link to kubernetes.io (e.g., http://kubernetes.io/community/) and keep this simpler. If some of this info isn't easily found there, we should fix that.

File: README.md [hunk 8]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
+Ask questions and help answer them on:
 
  * [Stack Overflow](http://stackoverflow.com/questions/tagged/kubernetes)
  * [Slack](http://slack.k8s.io/)
 
-### Want to contribute to Kubernetes?
+Attend our key events:
 
-If you're interested in being a contributor and want to get involved in developing Kubernetes, start in the [Kubernetes Developer Guide](docs/devel/README.md) and also review the [contributor guidelines](CONTRIBUTING.md).
+* [kubecon](http://events.linuxfoundation.org/events/kubecon)
```
Comment (bgrant0607) [inline README.md]: Hmm. I don't see info about kubecon on kubernetes.io. That would be useful to fix.

File: README.md [hunk 9]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
 
-Also, please see our [expectations](docs/devel/community-expectations.md) for members of
-the Kubernetes community.
+## Contribute
 
-### Support
+If you're interested in being a contributor and want to get involved in developing Kubernetes, get started with this light reading:
 
-While there are many different channels that you can use to get ahold of us, you can help make sure that we are efficient in getting you the help that you need.
+*  The community [expectations](docs/devel/community-expectations.md)
```
Comment (bgrant0607) [inline README.md]: @michelleN We'll need to update these links when the developer docs are moved.

File: README.md [hunk 10]
```diff
@@ -26,138 +23,131 @@ Kubernetes is:
...
+*  The [Kubernetes Developer Guide](docs/devel/README.md)
 
-If you need support, start with the [troubleshooting guide](docs/troubleshooting.md#getting-help) and work your way through the process that we've outlined.
+You will then most certainly gain a lot from joining a [SIG](https://github.com/kubernetes/community), attending the regular hangouts as well as the community [meeting](https://github.com/kubernetes/community/blob/master/community/README.md).
 
-That said, if you have questions, reach out to us one way or another.  We don't bite!
+If you have an idea for a new feature, see the [Kubernetes Features](https://github.com/kubernetes/features) repository for a list of features that are coming in new releases as well as details on how to propose one.
 
-### Community resources
+## Documentation
```
Comment (bgrant0607) [inline README.md]: There's a link to the doc site above, so I think we can just remove this section.

**Review by bgrant0607 — CHANGES_REQUESTED** [review]

**Comment by bgrant0607** [comment]
Thanks for the cleanup, @sebgoa!

File: README.md (hunk 9 above)
Comment (michelleN) [inline README.md]: Will do

**Comment by sebgoa** [comment]
Ok addressed all comments.

I kept some links in the community section, just made it a bit more compact. I will make sure to revisit once the kubernetes.io/community page gets in a better state.

Ideally I would like to add a simple build section once I get passed https://github.com/kubernetes/kubernetes/issues/36655 , showing how you can build k8s in two bash commands will be powerful.

**Comment by sebgoa** [comment]
I think we could also remove code-of-conduct.md since it is now a link to cncf. I am linking directly to it in the readme.

**Comment by bgrant0607** [comment]
@sebgoa code-of-conduct.md is currently part of the repo template. If we remove it here, we should make a similar change to the template. philips may have an opinion on that.

**Comment by bgrant0607** [comment]
Verify checks failed. 
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/36817/pull-kubernetes-verify/4754/

**Comment by bgrant0607** [comment]
LGTM other than the check failures.

**Comment by sebgoa** [comment]
ok tests are passing now. hadn't run munge-docs so fixed some broken links in docs/ docs/devel and examples/ folders.

**Comment by bgrant0607** [comment]
@saad-ali Would you mind if this were merged now?

**Review by bgrant0607 — APPROVED** [review]

**Comment by saad-ali** [comment]
> @saad-ali Would you mind if this were merged now?

Doc changes are ok for post-code-freeze merge.

**Comment by philips** [comment]
I am fine with the code-of-conduct move. I will do the same for the template project.

**Pull request merged**